plugins/ilert/README.md

```diff
@@ -0,0 +1,126 @@
+
+In detail this plugin provides:
+
+- Information details about the person on-call (all escalation levels of the current time)
+- A way to override the current on-call person
+- A list of active incidents
+- A way to trigger a new incident
+- A way to reassign/acknowledge/resolve an incident
+- A way to trigger an incident action
+- A way to trigger an immediate maintenance
+- A way to disable/enable an alert source
+- A list of uptime monitors
+
+## Setup instructions
+
+Install the plugin:
+
+```bash
+yarn add @backstage/plugin-ilert
+```
+
+Then make sure to export the plugin in your app's [`plugins.ts`](https://github.com/backstage/backstage/blob/master/packages/app/src/plugins.ts) to enable the plugin:
+
+```js
+export { plugin as ILert } from '@backstage/plugin-ilert';
+```
+
+Add it to the `EntityPage.tsx`:
+
+```ts
+import {
+  isPluginApplicableToEntity as isILertAvailable,
+  EntityILertCard,
+} from '@backstage/plugin-ilert';
+{
+  isILertAvailable(entity) && (
+    <Grid item md={6}>
+      <EntityILertCard />
+    </Grid>
+  );
+}
+```
+
+> To force an iLert card for each entity just add the `<EntityILertCard />` component. An instruction card will appear if no integration key is set.
+
+## Add iLert explorer to the app sidebar
+
+Modify your app routes in [`App.tsx`](https://github.com/backstage/backstage/blob/master/packages/app/src/App.tsx) to include the Router component exported by the plugin - for example:
+
+```tsx
+import { ILertPage } from '@backstage/plugin-ilert';
+<Routes>
+  // ...
+  <Route path="/ilert" element={<ILertPage />} />
+  // ...
+</Routes>;
+```
+
+Modify your sidebar in [`Root.tsx`](https://github.com/backstage/backstage/blob/master/packages/app/src/components/Root/Root.tsx) to include the icon component exported by the plugin - for example:
+
+```tsx
+import { ILertIcon } from '@backstage/plugin-ilert';
+<Sidebar>
+  // ...
+  <SidebarItem icon={ILertIcon} to="ilert" text="iLert" />
+  // ...
+</Sidebar>;
+```
+
+## Client configuration
+
+If you want to override the default URL for api calls and detail pages, you can add it to `app-config.yaml`.
+
+In `app-config.yaml`:
+
+```yaml
+ilert:
+  baseUrl: https://my-org.ilert.com/
+```
+
+## Providing the Authorization Header
+
+In order to make the API calls, you need to provide a new proxy config which will redirect to the [iLert API](https://api.ilert.com/api-docs/) endpoint. It needs an [Authorization Header](https://api.ilert.com/api-docs/#section/Authentication).
+
+Add the proxy configuration in `app-config.yaml`
+
+```yaml
+proxy:
+  ...
+  '/ilert':
+    target: https://api.ilert.com
+    allowedMethods: ['GET', 'POST', 'PUT']
+    allowedHeaders: ['Authorization']
+    headers:
+      Authorization:
+        $env: ILERT_AUTH_HEADER
+```
+
+Then start the backend, passing the token as environment variable:
+
+```bash
+$ ILERT_AUTH_HEADER='Basic <TOKEN>' yarn start
```

Comment:
we plan in the near future to introduce API tokens so that users will be able to use the bearer token instead of the basic auth, therefore the universal variable is used here. I will describe it better in the README.